pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/QuestionDao.java

```diff
@@ -417,11 +420,30 @@ default PicklistQuestion getPicklistQuestion(QuestionDto dto,
         List<PicklistGroup> groups = new ArrayList<>();
         List<PicklistOption> allOptions = new ArrayList<>();
 
+        PicklistOption option = null;
         // Put the ungrouped options in the list first.
         for (PicklistOptionDto optionDto : container.getUngroupedOptions()) {
-            allOptions.add(new PicklistOption(optionDto.getStableId(),
-                    optionDto.getOptionLabelTemplateId(), optionDto.getTooltipTemplateId(), optionDto.getDetailLabelTemplateId(),
-                    optionDto.getAllowDetails(), optionDto.isExclusive()));
+            if (CollectionUtils.isEmpty(optionDto.getNestedPicklistOptions())) {
+                option = new PicklistOption(optionDto.getStableId(),
+                        optionDto.getOptionLabelTemplateId(), optionDto.getTooltipTemplateId(), optionDto.getDetailLabelTemplateId(),
+                        optionDto.getAllowDetails(), optionDto.isExclusive());
+            } else {
+                //find any suboptions
+                if (CollectionUtils.isNotEmpty(optionDto.getNestedPicklistOptions())) {
```

Comment:
This `if` check looks redundant? This `else` block is for already for the not-empty branch...